src/backend/vulkan/src/lib.rs

```diff
@@ -321,13 +355,33 @@ impl hal::Instance<Backend> for Instance {
                 .unwrap_or(std::ptr::null_mut())
         });
 
+        // Pick the latest version of Vulkan available.
+        let api_version = match entry.try_enumerate_instance_version() {
+            // Vulkan 1.1+
+            Ok(Some(version)) => Version(version),
+
+            // Vulkan 1.0
+            Ok(None) => Version::V1_0,
+
+            // Ignore out of memory since it's unlikely to happen and `Instance::create` doesn't have a way to express it in the return value.
+            Err(err) if err == vk::Result::ERROR_OUT_OF_HOST_MEMORY => {
+                panic!("{}", err);
+            }
+
+            Err(_) => unreachable!(),
+        };
+
         let app_name = CString::new(name).unwrap();
         let app_info = vk::ApplicationInfo::builder()
             .application_name(app_name.as_c_str())
             .application_version(version)
             .engine_name(CStr::from_bytes_with_nul(b"gfx-rs\0").unwrap())
             .engine_version(1)
-            .api_version(vk::make_version(1, 0, 0));
+            .api_version(vk::make_version(
```

Comment:
any reason this isn't just `api_version`? are we trying to strip the patch index?

Reply:
You're totally right. I remember reading here (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/chap5.html#VkApplicationInfo) that the patch value is ignored, but I don't see any reason why we should strip it out. I'll just pass the `api_version` as is.